crates/aiken-lang/src/tipo/infer.rs

```diff
@@ -285,10 +286,51 @@ fn infer_definition(
                     });
                 }
 
+                let typed_other_fun = other_fun
+                    .map(|mut other| -> Result<TypedFunction, Error> {
+                        let params = params.into_iter().chain(other.arguments);
+                        other.arguments = params.collect();
+
+                        if let Definition::Fn(mut other_typed_fun) = infer_definition(
+                            Definition::Fn(other),
+                            module_name,
+                            hydrators,
+                            environment,
+                            tracing,
+                            kind,
+                        )? {
+                            if !other_typed_fun.return_type.is_bool() {
+                                return Err(Error::ValidatorMustReturnBool {
+                                    return_type: other_typed_fun.return_type.clone(),
+                                    location: other_typed_fun.location,
+                                });
+                            }
+
+                            other_typed_fun.arguments.drain(0..params_length);
+
+                            if other_typed_fun.arguments.len() < 2
+                                || other_typed_fun.arguments.len() > 3
```

Comment:
I think the check should be a bit more complicated here. As we can't have two functions with arity 2, or two functions with arity 3. So, we need yet-another-check after this one (and the similar one above for `typed_fun`) which ensures that the sum of their arity is `5`.

Reply:
Yes, I didn't get it it last night but I will do this kind of check today. At the end of I going to just make sure that the lengths aren't equal. And return an err with cool labels.